content/08-mongodb/14-mongodb-encryption.mdx

```diff
@@ -0,0 +1,94 @@
+
+## Types of encryption
+Encryption methods usually fall into two categories: symmetric-key encryption and asymmetric encryption. Their definitions are:
+- Symmetric Encryption - A method where the encryption key and the decryption key are the same (symmetrical)
+- Asymmetric Encryption - A method also known as public key cryptography that creates two different keys, a private key and a public key that are not identical. The public key can be shared with anyone, and the private key only with authorized people. The public key can be used to encrypt data that can only be decrypted by the private key.
+
+Both methods have some inherent vulnerabilities depending on the data status but cover up RBAC missed issues like networking snooping and data theft. This assurance is why MongoDB employs both methods across its databases, depending on the data state.  
+
+### Encryption in transit
+[Encryption in transit](/intro/database-glossary#encryption-transport) is securing data when it is in motion from one point to another. Before being sent, the data is encrypted for its “journey” and then decrypted and verified at the reception point. The best visualization is to imagine the data as money transported via an armored van from one bank branch to another to replenish the vault. 
+
+[MongoDB Atlas] supports [transport layer security (TSL)] and secure socket layer (SSL) for communications between application client and server and within intra-cluster communications by setting certificates for the servers. A certificate is an electronic document used to prove the validity of ownership of a public key. Certificates ensure data is encrypted for transport over a trusted network connection preventing network snooping activities such as packet sniffing or IP/DNS spoofing. 
+
+With the inclusion of a public key, we see that MongoDB utilizes asymmetric encryption to secure data in a state of motion. This is because of an inherent vulnerability in symmetric encryption for data in transit. The secret symmetric key must remain secure during the transmission. To secure this original key, you must encrypt it and create another key to decrypt. This cycle can continue infinitely, where every key in transmission is then reliant on another made key to secure it. Asymmetric encryption theoretically solves this insecure transmission problem with its public keys. 
+
+By encrypting with a public key, the sender’s message can only be decrypted by the intended recipient with their private key. Public key cryptography is advantageous in transit because there is no need to ever share private keys and the use of signatures ensures a recipient can verify that a message comes from a trusted sender. 
+
+The risk for servers relying on the asymmetric encryption is that the public key is not authentic and generated in bad faith or by a malicious entity. Anyone can access a public key to encrypt data, but only the private key owner can decrypt the data.
```

Comment:
I am just going to scrap this. I have been reading about the potential for compromised public keys whereas someone would replace Bob's key with their own generated key for example. It would get discovered quickly most likely, but that could already be too late. I am just going to speak instead to the disadvantage of asymmetric encryption taking longer.